multi_folder_download.py

```diff
@@ -0,0 +1,56 @@
+import argparse
+import os
+import subprocess
+import sys
+import time
+from datetime import datetime
+from distutils.util import strtobool
+
+import django
+from dateutil.relativedelta import relativedelta
+from dateutil.tz import tzutc
+from django.contrib.gis.geos import GEOSGeometry
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'geospaas_processing.settings')
+
+import geospaas_processing.downloaders as downloaders
+from geospaas.catalog.models import Dataset
+
+
+def main_function(ar):
+    if strtobool(ar.rel_time_flag):
+        designated_start_time = datetime.now().replace(tzinfo=tzutc())+ relativedelta(
+            hours=-int(ar.rel_time_lag))
+        designated_end_time = datetime.now().replace(tzinfo=tzutc())
+    else:
+        designated_start_time = datetime.strptime(ar.start_time, "%Y.%m.%d:%H").replace(
+            tzinfo=tzutc())
+        designated_end_time = datetime.strptime(ar.end_time, "%Y.%m.%d:%H").replace(tzinfo=tzutc())
+    downloaders.DownloadManager.MAX_DOWNLOADS = 500
+    download_manager = downloaders.DownloadManager(
+        download_directory=ar.mount_dir.rstrip(os.path.sep)+os.path.sep+ar.dir_struct.lstrip(
+            os.path.sep),
+        use_file_prefix=strtobool(ar.use_filename_prefix),
+        provider_settings_path=os.path.join(os.path.dirname(
+            __file__), ar.config_folder, 'provider_settings.yml'),
+        time_coverage_start__gte=designated_start_time,
+        time_coverage_end__lte=designated_end_time,
+        geographic_location__geometry__intersects=GEOSGeometry(ar.geometry),
+        dataseturi__uri__contains=ar.criteria)
```

Comment:
Make it generic and allow search not only by `dataseturi__uri`. For example if I want to search by source I want to call this script with option `-criterium source__instrument__short_name__icontains='SAR'`
We've discussed this how to implement that. criteria can be a dictionary that you generate from several `-c` options and then pass to DownloadManager as `**kwargs`

Reply:
Done in the next commit

Reply:
Done in the next commit.